src/hooks/useUpdateAssetOnchainBalance.js

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { getOnchainAssetBalance } from '../handlers/assets';
+import { dataUpdateAssets } from '../redux/data';
+import useAccountAssets from './useAccountAssets';
+import { logger } from '@rainbow-me/utils';
+
+export default function useUpdateAssetOnchainBalance() {
+  const { allAssets } = useAccountAssets();
+  const dispatch = useDispatch();
+
+  const useUpdateAssetOnchainBalance = useCallback(
+    async (assetToUpdate, accountAddress, successCallback) => {
+      const balance = await getOnchainAssetBalance(
+        assetToUpdate,
+        accountAddress
+      );
+      if (balance) {
+        if (balance.amount !== assetToUpdate.balance.amount) {
+          // Now we need to update the asset
+          // First in the state
+          successCallback({ ...assetToUpdate, balance });
+          // Then in redux
+          const allAssetsUpdated = allAssets.map(asset => {
```

Comment:
Just a note: if asset price updates come in from zerion, they'll overwrite the redux data. This is still probably useful though.

Reply:
Yes, I'm aware. The reason I did this was because it's the only way to update the balance in the currency selection display of the send flow without a massive refactor